src/streamlink/plugins/ard_live.py

```diff
@@ -1,56 +1,68 @@
 import re
+import json
+import logging
 
 from streamlink.plugin import Plugin
 from streamlink.plugin.api import validate
-from streamlink.stream import HLSStream, HDSStream
+from streamlink.stream import HLSStream
 from streamlink.compat import urljoin
-from streamlink.stream import HTTPStream
+from streamlink.exceptions import PluginError
+
+log = logging.getLogger(__name__)
 
 
 class ard_live(Plugin):
-    swf_url = "http://live.daserste.de/lib/br-player/swf/main.swf"
     _url_re = re.compile(r"https?://(www.)?daserste.de/", re.I)
-    _player_re = re.compile(r'''dataURL\s*:\s*(?P<q>['"])(?P<url>.*?)(?P=q)''')
+    _player_re = re.compile(r'''data-ctrl-player\s*=\s*"(?P<jsondata>.*?)"''')
     _player_url_schema = validate.Schema(
         validate.transform(_player_re.search),
         validate.any(
             None,
-            validate.all(validate.get("url"), validate.text)
-        )
+            validate.all(validate.get("jsondata"), validate.text)
+        ),
+        validate.transform(lambda v: json.loads(v.replace("'", '"'))),
```

Comment:
https://www.daserste.de/information/reportage-dokumentation/unser-wetter/sendung/wo-unser-wetter-entsteht-die-alpen-folge-1-100.html

`AttributeError: 'NoneType' object has no attribute 'replace'`

might need to check for `videos` in the __url_re_

https://www.daserste.de/information/reportage-dokumentation/unser-wetter/videos/wo-unser-wetter-entsteht-die-alpen-folge-1-video-100.html

Reply:
> AttributeError: 'NoneType' object has no attribute 'replace'

With me there is a Pluginerror:
`error: Unexpected stream type: 'https://pdvideosdaserste-a.akamaihd.net/int/2020/05/18/0b016d84-eec7-425e-9a1a-74631ccea882/320-1_678758.mp4'`

> might need to check for videos in the _url_re

I just got the test streams looked at and thought that the plugin is only for live streams, as the plugin name also says.

If also the videos are to be considered, then it would be better to rename the plugin in **daserste**?